src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -105,4 +105,7 @@
 
     @Query("select distinct participation from StudentParticipation participation left join fetch participation.results where participation.student.id = :#{#studentId} and participation.exercise in :#{#exercises}")
     List<StudentParticipation> findByStudentIdWithEagerResults(@Param("studentId") Long studentId, @Param("exercises") Set<Exercise> exercises);
+
+    @Query("select distinct p from StudentParticipation p left join fetch p.submissions s left join fetch s.result r where p.student.id = :#{#studentId} and p.exercise in :#{#exercises}")
+    List<StudentParticipation> findAllByStudentIdWithSubmissionsWithResult(@Param("studentId") Long studentId, @Param("exercises") Set<Exercise> exercises);
```

Comment:
- You can remove the joins when you use `@EntityGraph`
- The method name is not self explanatory as the participations for the student ids will not be retrieved if not part of the exercise
- For readability `...withEagerSubmissionsAndResults` is better

Reply:
I let the `@EntityGraph` annotation out for now and adjusted the method name as you both suggested.